lib/domain/use_cases/use_case_interface.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:dartz/dartz.dart';
+
+import '../entity/movie.dart';
+
+mixin GetPopularMoviesUseCase{
+
+  Future<EitherMovieAPI<List<MovieEntity>>> fetchPopularMovies();
+}
+
+mixin GetNowPlayingMoviesUseCase{
+
+  Future<EitherMovieAPI<List<MovieEntity>>> fetchNowPlayingMovies();
+}
+
+mixin GetTopRatedMoviesUseCase{
+
+  Future<EitherMovieAPI<List<MovieEntity>>> fetchTopRatedMovies();
+}
+
+mixin GetUpcomingMoviesUseCase{
+
+
+  Future<EitherMovieAPI<List<MovieEntity>>> fetchUpcomingMovies();
+}
+
+mixin GetGenreUseCase {
+
+  Future<EitherMovieAPI<Map<int,String>>> fetchGenres();
+}
+
+class Failure {
+  final int code;
+  final String message;
+
+  Failure(
+      this.code,
+      this.message,
+      );
+
+  @override
+  bool operator == (Object other) =>
+      // identical(this, other) ||
```

Comment:
why is this commented?

Reply:
I was testing if the other assertions worked properly.
Done